src/algebra/regular.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2021 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import algebra.group
+
+/-!
+# Regular elements
+
+We introduce left-regular, right-regular and regular elements.
+
+By definition, a regular element in a commutative ring is a non-zero divisor.
+Lemma `is_regular_of_integral_domain` shows that every non-zero element of an integral domain
+is regular.
+
+The final goal is to develop part of the API to prove, eventually, results about non-zero-divisors.
+-/
+variables {R : Type*}
+
+/-- A left-regular element is an element `c` such that multiplication on the left by `c`
+is injective on the left. -/
+def is_left_regular [has_mul R] (c : R) := function.injective ((*) c)
+
+/-- A right-regular element is an element `c` such that multiplication on the right by `c`
+is injective on the left. -/
```

Comment:
These docstrings have three lefts and two rights...

Reply:
Corrected, thanks!